api/streamableResponseWriter.go

```diff
@@ -13,15 +13,23 @@ type flushableWriter interface {
 	io.Writer
 	http.Flusher
 }
+
 type streamableResponseWriter struct {
-	flushableWriter
+	ctx    context.Context
+	writer flushableWriter
 }
 
 // Write flushes the data immediately after every write operation
 func (wf *streamableResponseWriter) Write(b []byte) (n int, err error) {
-	n, err = wf.flushableWriter.Write(b)
-	wf.Flush()
-	return n, err
+	select {
+	case <-wf.ctx.Done():
+		// Don't send anything if the context's finished. Pretend nothing happened.
```

Comment:
@ekinanp I'm not really sure this is always desirable, but it reproduces the behavior you had for sendPacket.

Reply:
Hmm, I guess not including it forces callers to think about proper cleanup, b/c if your streaming endpoint keeps writing it likely means that you're not closing a reader when you're supposed to. That's what led to `sendPacket` in Exec.

I guess it is better to remove it, even if it conceptually makes sense. Sorry for the extra time-waste.